### DISCUSSION TITLE
[Question]: Bundle a generic library with workers for NodeJS and Electron

### DISCUSSION THREAD

**Discussion by isasmendiagus** [webpack/webpack#13788] [Q&A]
### **Context**:
We have a central JS-based library that uses workers. We are trying to bundle the library with WebPack and use it among differents environments (NodeJS, Electron and also will be used to develop a plugin for VS Code).

### **Problem**:
The bundle is working in NodeJS so far so good. But when we try to import it into an electron application ([Electron-react-boilerplate](https://github.com/electron-react-boilerplate/electron-react-boilerplate)), we have some problems with the worker path and the import module system.

```
A JavaScript error occurred in the main process
Uncaught Exception:
/home/ubuntu/Projects/SCANOSS/scanoss-dt/src/main/scannerLib/electron-scanoss.js:670
    a.exports = (0, _module.createRequire)(import.meta.url)("fs");
                                                  ^^^^

SyntaxError: Cannot use 'import.meta' outside a module
    at wrapSafe (internal/modules/cjs/loader.js:1060:16)
```

**Steps to reproduce the scenario.**
1. Download and build the library: https://github.com/isasmendiagus/scanoss-scannerJS (our library uses workers) 
2. yarn install
3. yarn build
4. Integrate manually the bundles files into electron-react-boilerplate project. To do that: copy the bundles files in to the folder electron-react-boilerplate/src/main and try to import the bundled library from the code.

### **Questions**:
1. Is there a way to bundle a generic library that uses workers and works on all platforms (NodeJS and Electron)?
2. Is the webpack.config.js in our [repo](https://github.com/isasmendiagus/scanoss-scannerJS) correct?

**Other relevant information:**
webpack version: webpack version: 5.44.0
Node.js version: v14.17.0
Operating System: Ubuntu LTS 20.04
Additional tools: Yarn

**Comment by alexander-akait** [comment]
Please ask this in electron repo, you set https://github.com/isasmendiagus/scanoss-scannerJS/blob/main/webpack.config.js#L9, so `import.meta.url` should be supported, but looks like your env is not supported `import.meta.url` why? Due:

> SyntaxError: Cannot use 'import.meta' outside a module

electron thinking it is commonjs format, try change `filename: 'electron-scanoss.js'` on `filename: 'electron-scanoss.mjs',`

Please use https://github.com/webpack/webpack/discussions for questions in future
